engines/scc_proxy/lib/scc_proxy/engine.rb

```diff
@@ -24,66 +25,107 @@
   Net::HTTPRetriableError
 ].freeze
 
+INSTANCE_ID_KEYS = {
+  Amazon: 'instanceId',
+  Google: 'instance_id',
+  Microsoft: 'vmId'
+}.freeze
+
 # rubocop:disable Metrics/ModuleLength
 module SccProxy
   class << self
 
-    def headers(auth)
+    # rubocop:disable ThreadSafety/ClassAndModuleAttributes
+    attr_accessor :instance_id
+
+    # rubocop:enable ThreadSafety/ClassAndModuleAttributes
+
+    # rubocop:disable ThreadSafety/InstanceVariableInClassMethod
+    def headers(auth, params, logger)
+      @instance_id = if params && params.class != String
```

Comment:
I thought about this. Right now we do not enable `system_token` support at all right? I would keep it for now and not allow normal `system_token` handling.
On SCC side we are not ready yet, meaning we would need to change our implementation allowing the "new" case: BYOS system with normal system token.

As of now there is also no benefit over the existing implementation.
I advocate for not adding this now, see if we really need the token handling later and act accordingly if this will be the case.